Add tests for uniqueness constraints

robintw opened the pull request:
The PR to add uniqueness constraints to the reference tables (#354) was accidentally merged without any tests. This PR adds tests for these uniqueness constraints.

IanMayo commented:
@robintw - I think it's probably worth you double-checking that this PR matches our new understanding of uniqueness, please.

robintw commented:
@IanMayo I've fixed this to work with the new `level` parameter to `Privacy`. All of this still applies in terms of uniqueness as this PR only tests uniqueness for reference tables - I've updated the title to make this clear. There will be new tests added for uniqueness in the separate PR that adds unique constraints for the metadata tables.
